supportedBackends/google/pipelines/common/src/main/scala/cromwell/backend/google/pipelines/common/PapiPollResultMonitorActor.scala

```diff
@@ -0,0 +1,54 @@
+package cromwell.backend.google.pipelines.common
+
+import akka.actor.Props
+import cromwell.backend.google.pipelines.common.api.RunStatus
+import cromwell.backend.standard.pollmonitoring.{
+  AsyncJobHasFinished,
+  PollResultMessage,
+  PollResultMonitorActor,
+  ProcessThisPollResult
+}
+import cromwell.services.metadata.CallMetadataKeys
+
+import java.time.OffsetDateTime
+
+object PapiPollResultMonitorActor {
+  def props(tellMetadataFn: Map[String, Any] => Unit,
+            tellBardFn: (String, OffsetDateTime, Option[OffsetDateTime], OffsetDateTime) => Unit
+  ): Props = Props(new PapiPollResultMonitorActor(tellMetadataFn, tellBardFn))
+}
+
+class PapiPollResultMonitorActor(tellMetadataFn: Map[String, Any] => Unit,
+                                 tellBardFn: (String, OffsetDateTime, Option[OffsetDateTime], OffsetDateTime) => Unit
+) extends PollResultMonitorActor[RunStatus] {
+
+  override def extractEarliestEventTimeFromRunState(pollStatus: RunStatus): Option[OffsetDateTime] =
+    pollStatus.eventList.minByOption(_.offsetDateTime).map(e => e.offsetDateTime)
+  override def extractStartTimeFromRunState(pollStatus: RunStatus): Option[OffsetDateTime] =
+    pollStatus.eventList.collectFirst {
+      case event if event.name == CallMetadataKeys.VmStartTime => event.offsetDateTime
+    }
+
+  override def extractEndTimeFromRunState(pollStatus: RunStatus): Option[OffsetDateTime] =
+    pollStatus.eventList.collectFirst {
+      case event if event.name == CallMetadataKeys.VmEndTime => event.offsetDateTime
+    }
+
+  override def tellMetadata(metadata: Map[String, Any]): Unit = tellMetadataFn(metadata)
```

Comment:
Can we give this actor a handle to the metadata service registry rather than having all the backends pass in (I assume identical?) implementations?

Similar question for Bard - can we put the actual message-passing to the Bard service in the parent `PollResultMonitorActor` and figure out the minimum logic that needs to be per-backend?

Reply:
That is certainly possible, but I'm not sure if that's a big improvement since we would also need to pass in a lot of information about the particular task (runtime attributes, job descriptor, etc...) in order to have all the information necessary to send the message in the right way.

Reply:
Yeah, I see what you mean, for Bard. Maybe we could have a method that takes the current inputs and creates and returns a `TaskSummaryEvent`, which the `PollMonitorActor` knows what to do with?

Reply:
Updated branch to do something along these lines. We no longer pass in callbacks from individual backends, but instead pass in some extra data so that the poll monitors can call their own implementations of `tellMetadata` and `tellBard` instead. This allowed us to remove `tellBard` from elsewhere in the codebase.